### ISSUE TITLE
1.2.0: build issue

### ISSUE THREAD

**Issue by kloczek** [rrthomas/libpaper#19]
Looks like there is no dist tar ball with integrated gnulib and ..
```console
[tkloczko@devel-g2v libpaper-1.2.0]$ ./bootstrap
bootstrap: warning: Consider installing git-merge-changelog from gnulib.
bootstrap: running: /usr/bin/libtoolize --quiet
bootstrap: running: git clone --shallow-since=2021-01-13 'git://git.sv.gnu.org/gnulib.git' 'gnulib'
Cloning into 'gnulib'...
fatal: unable to connect to git.sv.gnu.org:
git.sv.gnu.org[0: 209.51.188.168]: errno=Connection timed out
git.sv.gnu.org[1: 2001:470:142::168]: errno=Network is unreachable
```
Issue:
- IMO it would be relly good to detach completly libpaper from gnulib. All OSess now fully supoorts basic ANSI and POSIX APIs. Witout gnulib it would be possible to simplifuy many parts of the code. It would allow simplify source tree and dtach it from externam submodule. Without that submodule relation as dist tar ball could be used autogenerated by github from got tag tar ball like https://github.com/rrthomas/libpaper/archive/refs/tags/v1.2.0.tar.gz
- lates autoreconf from autoconf 2.71 shows warning
```console
[tkloczko@devel-g2v libpaper-1.2.0]$ autoreconf -fiv
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force -I m4 -I src/m4
autoreconf: configure.ac: tracing
autoreconf: running: libtoolize --copy --force
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'build-aux'.
libtoolize: copying file 'build-aux/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: aclocal --force -I m4 -I src/m4
autoreconf: running: /usr/bin/autoconf --force
configure.ac:60: warning: The macro `AC_PROG_CC_C99' is obsolete.
configure.ac:60: You should run autoupdate.
./lib/autoconf/c.m4:1659: AC_PROG_CC_C99 is expanded from...
configure.ac:60: the top level
configure.ac:65: warning: The macro `AC_GNU_SOURCE' is obsolete.
configure.ac:65: You should run autoupdate.
./lib/autoconf/specific.m4:312: AC_GNU_SOURCE is expanded from...
configure.ac:65: the top level
configure.ac:65: warning: AC_COMPILE_IFELSE was called before AC_USE_SYSTEM_EXTENSIONS
./lib/autoconf/specific.m4:312: AC_GNU_SOURCE is expanded from...
configure.ac:65: the top level
configure.ac:65: warning: AC_LINK_IFELSE was called before AC_USE_SYSTEM_EXTENSIONS
./lib/autoconf/specific.m4:312: AC_GNU_SOURCE is expanded from...
configure.ac:65: the top level
configure.ac:65: warning: AC_CHECK_INCLUDES_DEFAULT was called before AC_USE_SYSTEM_EXTENSIONS
./lib/autoconf/specific.m4:312: AC_GNU_SOURCE is expanded from...
configure.ac:65: the top level
configure.ac:142: warning: The macro `AC_CONFIG_HEADER' is obsolete.
configure.ac:142: You should run autoupdate.
./lib/autoconf/status.m4:719: AC_CONFIG_HEADER is expanded from...
configure.ac:142: the top level
autoreconf: running: /usr/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:60: installing 'build-aux/compile'
configure.ac:26: installing 'build-aux/missing'
lib/Makefile.am: installing 'build-aux/depcomp'
automake: error: cannot open < libgnu/Makefile.gnulib: No such file or directory
autoreconf: error: automake failed with exit status: 1
```

**Comment by rrthomas** [comment]
As per `README`, no need to bootstrap a source distribution.

**Comment by rrthomas** [comment]
As you can see in https://github.com/rrthomas/libpaper/blob/master/bootstrap.conf , gnulib is used only for non-POSIX functions and other functionality (crucially to libpaper, the `relocatable-perl` module). (The use of `free-posix` and `realloc-posix` is to work around a build problem.)

**Comment by kloczek** [comment]
That is thae isse that libpaper it is not so complicated that it needs to use non-POSIX functions ..

**Comment by rrthomas** [comment]
Ha! Well, they are not very complicated functions: `getline`, `strcase` and `strndup`. Still, they are all easy to get wrong, and I don't see much point in duplicating work in their implementation when I can use tried-and-tested definitions that are shared with other projects. Gnulib is well-maintained, and benefits from frequent fixes and improvements.

**Comment by kloczek** [comment]
Don't get me wrong .. I'm not pointing on that to remove that yesterday or tody 😋 
Using libgnu was OK +15 years ago but if you will look around in mean time some projects managed to drop use of it and raplace by something else😃

**Comment by rrthomas** [comment]
Unfortunately(!), I've been adding gnulib to projects in the last 15 years, and indeed it's more useful today than it was when I started. I have thought about what it would take to remove it from one or two specific projects, and the answer is that it would be a lot of work, whether to find replacements for the various non-standard APIs that it provides, or for the other non-code tooling that it offers (such as relocatability). For portable code, GLib may be a reasonable base. For portable build systems (replacing also GNU autotools) I'm less sure: all the alternatives that I've seen seem to require system-specific code in the build system, whereas I typically have none in most projects and just a handful of lines for Windows in one or two. Moving away from C makes things much easier, but of course that is often not practical (for example, I did rewrite libpaper entirely in Perl, but that would have required substantial patches to its existing C clients, and in the end I gave up and went back to C). I have also used Vala for some projects (e.g. I rewrote GNU Zile in Vala), which would make it easier to move away from autotools and gnulib; but "easier", not "easy".
